cmake/platforms/macos.cmake

```diff
@@ -0,0 +1,92 @@
+message(STATUS "Building NovelRT for macOS")
+
+include(NovelRTBuildDeps)
+
+#
+# Platform-specific changes
+#
+set(CMAKE_OSX_DEPLOYMENT_TARGET "10.15" CACHE STRING "Minimum macOS / OS X deployment version")
+set(CMAKE_MACOSX_RPATH 1)
+set(CMAKE_SKIP_BUILD_RPATH FALSE)
+set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
+set(CMAKE_INSTALL_RPATH "@loader_path/")
+set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+list(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "@loader_path/" isSystemDir)
+if("${isSystemDir}" STREQUAL "-1")
+  set(CMAKE_INSTALL_RPATH "@loader_path/")
+endif("${isSystemDir}" STREQUAL "-1")
+
+#
+# Dependency checks
+#
+find_package(spdlog ${NOVELRT_SPDLOG_VERSION} REQUIRED)
+find_package(TBB ${NOVELRT_ONETBB_VERSION} CONFIG REQUIRED)
+find_package(Microsoft.GSL ${NOVELRT_GSL_VERSION} REQUIRED)
+find_package(stduuid CONFIG REQUIRED)
+find_package(glfw3 ${NOVELRT_GLFW_VERSION} CONFIG REQUIRED)
+find_package(glm REQUIRED)
+find_package(Ogg ${NOVELRT_OGG_VERSION} REQUIRED)
+find_package(OpenAL CONFIG REQUIRED)
+find_package(Opus REQUIRED)
+find_package(Vorbis ${NOVELRT_VORBIS_VERSION} REQUIRED)
+find_package(flac ${NOVELRT_FLAC_VERSION} REQUIRED)
+find_package(SndFile REQUIRED)
+find_package(nlohmann_json REQUIRED)
+find_package(Vulkan ${NOVELRT_VULKAN_VERSION} REQUIRED COMPONENTS MoltenVK)
```

Comment:
The duplication of this might be error-prone later down the line, though I can't really think of anything better myself right now, admittedly.

Reply:
The duplication is purely because we dont have platform-specific dependencies - I'm expecting this to sort itself as we introduce them